src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -9980,23 +10092,76 @@ and GenImplFileContents cenv cgbuf qname lazyInitInfo eenv mty def =
         let _eenvEnd = GenModuleOrNamespaceContents cenv cgbuf qname lazyInitInfo eenv def
         ())
 
+and CodeGenInitMethod cenv (cgbuf: CodeGenBuffer) eenv tref (codeGenInitFunc: CodeGenBuffer -> IlxGenEnv -> unit) m =
+
+    // Generate the declarations in the module and its initialization code
+    let _, body =
+        CodeGenMethod cenv cgbuf.mgbuf ([], eenv.staticInitializationName, eenv, 0, None, codeGenInitFunc, m)
+
+    if CheckCodeDoesSomething body.Code then
+        // We are here because the module we just grabbed has an interesting static initializer
+        let feefee, seqpt =
+            if body.Code.Instrs.Length > 0 then
+                match body.Code.Instrs[0] with
+                | I_seqpoint sp as i -> [ FeeFeeInstr cenv sp.Document ], [ i ]
+                | _ -> [], []
+            else
+                [], []
+
+        let ilDebugRange = GenPossibleILDebugRange cenv m
+
+        // Call global file initializer
+        match eenv.initClassFieldSpec with
+        | Some fs -> cgbuf.mgbuf.AddExplicitInitToCctor(tref, fs.Force(), ilDebugRange, eenv.imports, feefee, seqpt)
+        | None -> ()
+
+        // Add code to invoke envinner's class static initializer
+        let access = ComputeMemberAccess true taccessInternal eenv.realsig
+
+        let ilBody = MethodBody.IL(InterruptibleLazy.FromValue body)
+        let ilReturn = mkILReturn ILType.Void
+
+        let method =
+            mkILNonGenericStaticMethod (eenv.staticInitializationName, access, [], ilReturn, ilBody)
```

Comment:
I think the method should be generated with the MethodAttributes.SpecialName on it so that tooling knows to hide it. There is a `methodDef.WithSpecialName()`` to do that.

Reply:
Good call

Reply:
done